### PR TITLE
Part 1 and Part 2 (and a tiny bit of DDA)

### PR DESCRIPTION
**Pull request by ShutoAraki**
The `main.cpp` file should test all possible corner cases for `Color` and `Raster` class methods.

@sonegishi Should we set up a CI platform or some sort of automatic testings for every commit?

Questions:
1. Is it okay to do `#include "Raster.cpp"` in `main.cpp`? I feel like I should only include header files but what if you wanted to separate other `.cpp` files?
2. Why doesn't the following work?
```
#define WIDTH 200
#define HEIGHT 200
... whatever ...
Raster raster(WIDTH, HEIGHT, Red);
```
but `Raster raster(200, 200, Red);` works...

### CODE REVIEW

**Comment by sonegishi** [comment]
> Should we set up a CI platform or some sort of automatic testings for every commit?
@ShutoAraki We could but not necessarily. If you wanna make sure that every commit won't have a side effect on other tests, you might wanna have one.

**Comment by sonegishi** [comment]
> Is it okay to do #include "Raster.cpp" in main.cpp? I feel like I should only include header files but what if you wanted to separate other .cpp files?

@ShutoAraki I personally don't think it's okay to directly import `.cpp` files in `main.cpp`, and we should only include header files for sure. What do you mean by `separate other .cpp files`?

**Comment by sonegishi** [comment]
> Why doesn't the following work?

@ShutoAraki What does it return when you run the code?

**Comment by ShutoAraki** [comment]
@sonegishi I think I figured it out

1. All I had to do was changing `.cpp` into `.h` in the `#include` and then run `g++ main.cpp Raster.cpp Color.cpp -o main`!!

2. I don't know why this was an issue... It works fine now lol

**Comment by ShutoAraki** [comment]
> > Should we set up a CI platform or some sort of automatic testings for every commit?
> 
> @ShutoAraki We could but not necessarily. If you wanna make sure that every commit won't have a side effect on other tests, you might wanna have one.

I wanna learn DevOps so I wanna do it just for fun lol Travis CI or CircleCI or whatever

**Comment by sonegishi** [comment]
> I wanna learn DevOps so I wanna do it just for fun lol Travis CI or CircleCI or whatever

@ShutoAraki You could try either TravisCI or CircleCI. I prefer CircleCI, but you can choose either. It's pretty easy to implement. Which unit test are we using btw?

**Comment by ShutoAraki** [comment]
> > I wanna learn DevOps so I wanna do it just for fun lol Travis CI or CircleCI or whatever
> 
> @ShutoAraki You could try either TravisCI or CircleCI. I prefer CircleCI, but you can choose either. It's pretty easy to implement. Which unit test are we using btw?

Ok I'll look into CircleCI. Can you write some test file(s)?

File: Raster.cpp [hunk 1]
```diff
@@ -0,0 +1,111 @@
+#include <iostream>
+#include <cmath>
+#include <fstream>
+#include <new>
```
Comment (sonegishi) [inline Raster.cpp]: Do we need this?

Comment (ShutoAraki) [reply Raster.cpp]: Yeah we need `iostream` for printing errors, `cmath` for `round` function, `fstream` to create a PPM file, and `new` for `nothrow` dynamic memory allocation.

Comment (sonegishi) [reply Raster.cpp]: I was actually specifically asking about `new`, but okay!
